share/python/g3mp2.py

```diff
@@ -0,0 +1,685 @@
+            H298_sum_atoms += h298_atom
+
+            d0, d298 = dHf_one_atom(Z)              # in kcal/mol
+            dhf0_sum_atoms += d0
+            dhf298_sum_atoms += d298
+
+        dhf0 = (_E0 - E0_sum_atoms) * psi_hartree2kcalmol + dhf0_sum_atoms
+        dhf298 = (_H298 - H298_sum_atoms) * psi_hartree2kcalmol + dhf298_sum_atoms
+
+        # Eatomization = -(_H298 - H298_sum_atoms) * psi_hartree2kcalmol
+
+        return dhf0, dhf298
+
+    # ______________________________________________________
+    # _____________________      ___________________________
+    # _____________________ MAIN ___________________________
+    # ______________________________________________________
+
+    nAtoms = 0
+    charge = 0
+    multiplicity = 1
+
+    kwargs = p4util.kwargs_lower(kwargs)
+    debug('%s: %d args: %s' %
+          (name, len(kwargs), ','.join([arg for arg in kwargs])))
+
+    flag_list = ["verbose", "qcisdt", "debug", "help"]
+    arg_list = list(kwargs)
+
+    if "help" in kwargs:
+        g3mp2_usage()
+        return 0.0      # quit
+
+    for arg in arg_list:
+        if arg in flag_list:
+
+            if arg == 'verbose':
+                g3mp2_verbose_flag = kwargs.pop('verbose', True)
+
+            if arg == 'qcisdt':
+                use_QCISDT = kwargs.pop('qc == dt', True)
+
+            if arg == 'debug':
+                kwargs.pop('debug')
+                verbose()
+                set_debug(1)
+
+    mol = psi4.get_active_molecule()
+    g3mp2_name = mol.name()
+    nAtoms = mol.natom()
+    charge = mol.molecular_charge()
+    multiplicity = mol.multiplicity()
+
+    # debug("name: \"%s\" charge = %d multiplicity = %d." % (g3mp2_name, charge, multiplicity))
+    # debug("There are %d atoms, Z = [%s]" %
+    #     (nAtoms, join([str(int(mol.Z(i))) for i in range(nAtoms)])))
+
+    if multiplicity > 1:
+        if use_QCISDT is True:      # UHF-QCISD(T) unsupported
+            report("\n")
+            report("g3mp2: QCISD(T) doesn't support species with multiplicity > 1.")
+            report("g3mp2: Use the default CCSD(T) if acceptable.\n")
+            psi4.clean()
+            raise ValidationError("UHF unsupported for QCISD(T)\n")
+
+        else:
+            psi4.set_global_option('REFERENCE', 'UHF')
+
+    else:
+        psi4.set_global_option('REFERENCE', 'RHF')
+
+    # stash user options:
+    optstash = p4util.OptionsState(
+        ['FNOCC', 'COMPUTE_TRIPLES'],
+        ['FNOCC', 'COMPUTE_MP4_TRIPLES'],
+        ['FREEZE_CORE'],
+        ['MP2_TYPE'],
+        ['SCF', 'SCF_TYPE'])
+
+    # frobnicate the scf_type
+    # DF diverges from PK | DIRECT beyond 1.0E-04 Ha ~= 1/15 kcal
+    # OUT-OF-CORE:  slow
+    # psi4.set_local_option('SCF', 'SCF_TYPE', 'DIRECT')
+    psi4.set_local_option('SCF', 'SCF_TYPE', 'PK')
+
+    # optimize geometry at RHF/6-31G*
+    say('HF optimize.')
+
+    if nAtoms > 1:
+        psi4.set_global_option('BASIS', '6-31G(D)')
+        optimize('scf')
```

Comment:
Not to seem stone-headedly oblivious,  but if **molecule=<current molecule>** is passed explicitly, then 'molecule' should have been popped out of **kwargs beforehand.

Complaints from driver.py about multiple keyword values are instructive on this point.

Thank you all for your constructive remarks.  A corrected release should appear within a day.